### PR TITLE
Users should be able to Signup on Authors Haven

### PR DESCRIPTION
**Pull request by udochukwu**
**What does this PR do?**

- Users should be able to Signup and or Register using Sign up forms on the signup route.

**Description of Task to be completed?**

- Add signup feature

**What are the relevant pivotal tracker stories?**

- https://www.pivotaltracker.com/story/show/160451454

**Screenshots (if appropriate)**

- N/A

**Questions**

- N/A

### CODE REVIEW

File: src/containers/Signup/Signup.js [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+   * @returns {jsx} - jsx
+   * @memberof Signup
+   */
+  render() {
+    const { email, username, password } = this.state;
+    const { loading, success } = this.props;
+    return (
+    <Modal id='signupModal' className="center-align">
+      {
+        success
```
Comment (udochukwu) [inline src/containers/Signup/Signup.js]: Show success message on successful signup

File: src/containers/Signup/Signup.js [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        && <div id="signupSuccessCard">
+          <i className="fas fa-check-circle fa-5x"></i>
+          <h5>Registration was successful</h5>
+          <p>To proceed, Please check your email and verify your account</p>
+          <Button waves='light' onClick={this.closeModal}> OK</Button>
+        </div>
+      }
+      {
```
Comment (udochukwu) [inline src/containers/Signup/Signup.js]: hide signup form on successful signup

Comment (udochukwu) [inline src/containers/Signup/Signup.js]: hide signup form on successful signup

File: src/containers/Signup/Signup.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+   * @description - This method renders the jsx for this component
+   * @returns {jsx} - jsx
+   * @memberof Signup
+   */
+  render() {
+    const { email, username, password } = this.state;
+    const { loading, success } = this.props;
+    return (
+    <Modal id='signupModal' className="center-align">
+      {
```
Comment (udochukwu) [inline src/containers/Signup/Signup.js]: Show success message on successful signup

Comment (udochukwu) [inline src/containers/Signup/Signup.js]: hide signup form on successful signup

File: src/containers/Signup/Signup.js [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+        success
+        && <div id="signupSuccessCard">
+          <i className="fas fa-check-circle fa-5x"></i>
+          <h5>Registration was successful</h5>
+          <p>To proceed, Please check your email and verify your account</p>
+          <Button waves='light' onClick={this.closeModal}> OK</Button>
+        </div>
+      }
```
Comment (udochukwu) [inline src/containers/Signup/Signup.js:80]: Show success message on successful signup

File: src/containers/Signup/Signup.js [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+      {
+      !success
+      && <div>
```
Comment (udochukwu) [inline src/containers/Signup/Signup.js:83]: hide signup form on successful signup

File: src/containers/Header/Header.js [hunk 6]
```diff
@@ -180,7 +189,11 @@ class Header extends Component {
                 </a>
               </li>
               <li>
-                <a href="#" className="sign-up">Sign Up</a>
+                <a
+                  onClick={this.onSignupClicked.bind(this)}
```
Comment (s-akinrele) [inline src/containers/Header/Header.js]: move this to a constructor

**Review by s-akinrele — CHANGES_REQUESTED** [review]
As discussed

**Review by s-akinrele — APPROVED** [review]

**Pull request merged**